### PR TITLE
Do not serialize two properties when subscriptionRequired is false

### LINKED ISSUE #200: Cant deploy products template
It is a bug in the creation of products template.

I use the create tool, and set up some products. 

i set up one like this in yaml:
 - displayName: someName
    description: Some Description
    terms: https://someUrl.com
    subscriptionRequired: false
    state: Published
    policy: policies/generic/empty-policy.xml

notice that i generate this without:
subscriptionRequired: false
    approvalRequired: false
    subscriptionsLimit: 0

When i try to deploy i get this from Azure CLI:
Deployment failed. Correlation ID: 070483cd-7075-4d53-a0bb-88b5a8ab1d27. {

  "error": {

    "code": "ValidationError",

    "message": "Cannot provide values for approvalRequired and subscriptionsLimit when subscriptionRequired is set to false in the request payload",

    "details": null

  }

}

When i check the products-template.json, this values, that i left out is still provided in the template. but they should not be in there..

### PR DESCRIPTION
**Pull request by miaojiang**
This fixes #200

### CODE REVIEW

File: src/APIM_ARMTemplate/apimtemplate/Common/TemplateModels/ProductsTemplateResource.cs [hunk 1]
```diff
@@ -15,5 +15,16 @@ public class ProductsTemplateProperties
         public int subscriptionsLimit { get; set; }
         public string state { get; set; }
         public string displayName { get; set; }
+
+        // This properties should not be serialized if subscriptionRequired is false
+        public bool ShouldSerializeapprovalRequired(){
```
Comment (Frederic-roamler) [inline src/APIM_ARMTemplate/apimtemplate/Common/TemplateModels/ProductsTemplateResource.cs]: I can't find where this property is used but had the same issue and solved it by #197

**Comment by miaojiang** [comment]
The fix would solve the problem. However, if subscriptionRequired is false, and approvalRequired/subscriptionLimit is provided in the YAML, I think it would still result in invalid ARM template. 

The proper logic is to not include approvalRequired and subscriptionLimit in the ARM template, if subscriptionRequired is false.

File: src/APIM_ARMTemplate/apimtemplate/Common/TemplateModels/ProductsTemplateResource.cs (hunk 1 above)
Comment (miaojiang) [reply src/APIM_ARMTemplate/apimtemplate/Common/TemplateModels/ProductsTemplateResource.cs]: Please see my comment in #197

Comment (lucashuet93) [reply src/APIM_ARMTemplate/apimtemplate/Common/TemplateModels/ProductsTemplateResource.cs]: Both approaches will work but require additions. In both cases, if subscriptionsLimit and approvalRequired are provided in the creatorConfig then they will be written to the generated templates, causing errors. That behavior must be prevented. With @miaojiang changes, these methods will need to be called prior to serialization and ensure that subscriptionsLimit and approvalRequired are not serialized in the event that subscriptionsRequired is false. With @Frederic-roamler changes, the same is true - those two properties must be nullified forcefully in the event that the user provides values for them while setting subscriptionsRequired to false. I'm requesting changes from both PRs.

**Comment by lucashuet93** [comment]
Closing as same behavior was added in pull request #197

**Pull request closed without merging**